lib/src/blaze/block_witness_data.rs

```diff
@@ -725,7 +727,12 @@ impl BlockAndWitnessData {
                     yield_now().await;
                     blocks = self.blocks_in_current_batch.read().await;
                 }
-                self.sync_status.write().await.witnesses_updated = top_block - bottom_block - i;
+                self.sync_status.write().await.witnesses_updated.insert(
+                    <D::Bundle as crate::wallet::traits::Bundle<D, ChainType>>::Spend::wallet_nullifier(
```

Comment:
`Spend::wallet_nullifier` superficially looks like an AssType instead of a method.   Can we rename it to `get_wallet_nullfier`?

Reply:
Let's revisit the name `WitnessCache::top_height`.

Reply:
Maybe for another time but "MAX_REORG" and "REORG_BUFFER_OFFSET" are confusingly similar concepts.

Reply:
Why convert into and out of a 'fixed_size_buffer' if the eventual structure is a vec?

https://github.com/zingolabs/zingolib/pull/202/files#diff-fe24446938f05b862f0ff1d05aa869ec72d727a008b5c335db2cb868768b8a68R708

Reply:
Yikes!   `yield_now`!  

https://github.com/zingolabs/zingolib/pull/202/files#diff-fe24446938f05b862f0ff1d05aa869ec72d727a008b5c335db2cb868768b8a68R693

Inside `wait_for_first_block`...  is it  likely that this is a bottleneck since it busy-waits in the case of 0-ready-threads?

https://doc.rust-lang.org/std/thread/fn.yield_now.html

Reply:
> `Spend::wallet_nullifier` superficially looks like an AssType instead of a method. Can we rename it to `get_wallet_nullfier`?

I think the snake_case name makes this clear enough.. But.. This doesn't need to be an associated fn, I think a `from` impl is a better solution anyway. Will make a pr for that, and see if I can clean up some other traits as well.

Reply:
> Maybe for another time but "MAX_REORG" and "REORG_BUFFER_OFFSET" are confusingly similar concepts.

Good point. This is worth thinking about, and perhaps consolidating.

Reply:
> Why convert into and out of a 'fixed_size_buffer' if the eventual structure is a vec?
> 
> https://github.com/zingolabs/zingolib/pull/202/files#diff-fe24446938f05b862f0ff1d05aa869ec72d727a008b5c335db2cb868768b8a68R708

I suspect the fixed_size_buffer stuff is to avoid heap_allocation/resizing during sync for efficiency purposes, but it's a holdover from zecwallet that I haven't investigated much.